firefox_ui_tests/functional/security/test_no_certificate.py

```diff
@@ -0,0 +1,82 @@
+
+        self.url = self.marionette.absolute_url('layout/mozilla.html')
+
+        self.identity_popup = self.browser.navbar.locationbar.identity_popup
+
+    def tearDown(self):
+        try:
+            self.identity_popup.close(force=True)
+            self.windows.close_all([self.browser])
+        except NoSuchElementException:
+            # TODO: A NoSuchElementException may be thrown here when the test is skipped
+            # as under xvfb.
+            pass
+        finally:
+            FirefoxTestCase.tearDown(self)
+
+    @skip_if_e10s
+    @skip_under_xvfb
+    def test_no_certificate(self):
+        with self.marionette.using_context('content'):
+            self.marionette.navigate(self.url)
+
+        # Check the favicon
+        # TODO: find a better way to check, e.g., mozmill's isDisplayed
+        self.assertEqual(self.browser.navbar.locationbar.favicon.get_attribute('hidden'), 'false',
+                         'The globe favicon is visible')
+
+        # Check that the identity popup organization label is blank
+        self.assertEqual(self.identity_popup.organization_label.get_attribute('value'), '',
+                         'The favicon has no label')
+
+        self.identity_popup.box.click()
+        Wait(self.marionette).until(lambda _: self.identity_popup.is_open)
+
+        # Check the idenity popup doorhanger
+        self.assertEqual(self.identity_popup.popup.get_attribute('className'),
+                         'unknownIdentity', 'The Larry UI is unknown (aka Grey)')
+
+        # Open the Page Info window by clicking the More Information button
+        page_info = self.browser.open_page_info_window(
+            lambda _: self.identity_popup.more_info_button.click())
+
+        try:
+            # Verify that the current panel is the security panel
+            self.assertEqual(page_info.deck.selected_panel, page_info.deck.security,
+                             'The Security tab is selected by default')
+
+            # Check the domain listed on the security panel contains the url's host name
+            self.assertIn(urlparse(self.url).hostname,
+                          page_info.deck.security.domain.get_attribute('value'),
+                          'The domain label contains the host name in the url')
+
+            # Check the owner label equals localized 'securityNoOwner'
+            self.assertEqual(page_info.deck.security.owner.get_attribute('value'),
+                             page_info.get_property('securityNoOwner'),
+                             'The owner label equals the localized "securityNoOwner"')
+
+            # Check the verifier label equals localized 'notset'
+            self.assertEqual(page_info.deck.security.verifier.get_attribute('value'),
+                             page_info.get_property('notset'),
+                             'The verifier label equals the localized "notset"')
+
+        finally:
+            page_info.close()
+            self.browser.focus()
```

Comment:
I think we can really remove the try/finally given that we close all windows in tearDown anyway. See PR #131 how @chmanchester is doing it which is indeed cleaner, and we are not dependent on closing the page info window because its at the end of the test anyway.